static/src/javascripts/projects/common/modules/avatar/api.ts

```diff
@@ -0,0 +1,49 @@
+import config from 'lib/config';
+
+const apiUrl = `${String(config.get('page.avatarApiUrl'))}/v1`;
+const staticUrl = `${String(config.get('page.avatarImagesUrl'))}/user`;
+
+const request = (
+	method: string,
+	path: string,
+	data?: FormData,
+): Promise<Response> => {
+	const url = apiUrl + path;
+
+	if (method === 'POST') {
+		if (!data) throw new Error('POST error: No data provided.');
+
+		return fetch(url, {
+			method,
+			body: data,
+			mode: 'cors',
+			credentials: 'include',
+		}).then((resp) => resp.json() as Promise<Response>);
+	}
+
+	return fetch(url, {
+		method,
+		mode: 'cors',
+		credentials: 'include',
+	});
+};
+
+// A user's 'active' avatar is only available to signed-in users as it
+// includes avatars in a pre-mod state.
+const getActive = (): Promise<Response> =>
+	request('GET', '/avatars/user/me/active');
+
+const updateAvatar = (data: FormData): Promise<Response> =>
+	request('POST', '/avatars', data);
```

Comment:
```suggestion

// 2021-04-06: updating avatar should only be done from manage.theguardian.com – this can likely be removed.
const updateAvatar = (data: FormData): Promise<Response> =>
	request('POST', '/avatars', data);
```